src/remote.ts

```diff
@@ -34,13 +34,28 @@ export class NoSSHGatewayError extends Error {
 
 export const SSH_DEST_KEY = 'ssh-dest:';
 
-export function getGitpodRemoteWindowConnectionInfo(context: vscode.ExtensionContext): { remoteAuthority: string; connectionInfo: SSHConnectionParams } | undefined {
+type SSHConnType = 'local-app' | 'local-ssh' | 'ssh-gateway';
+export interface RemoteConnectionInfo {
+    remoteAuthority: string;
+    connectionInfo: SSHConnectionParams;
+    connType: SSHConnType;
+}
+
+export function getGitpodRemoteWindowConnectionInfo(context: vscode.ExtensionContext): RemoteConnectionInfo | undefined {
     const remoteUri = vscode.workspace.workspaceFile || vscode.workspace.workspaceFolders?.[0].uri;
     if (vscode.env.remoteName === 'ssh-remote' && context.extension.extensionKind === vscode.ExtensionKind.UI && remoteUri) {
         const [, sshDestStr] = remoteUri.authority.split('+');
+        const sshDestInfo = JSON.parse(Buffer.from(sshDestStr, 'hex').toString()) as { hostName: string; user?: string };
         const connectionInfo = context.globalState.get<SSHConnectionParams>(`${SSH_DEST_KEY}${sshDestStr}`);
         if (connectionInfo) {
```

Comment:
doesn't `connectionInfo` already have host?

Reply:
> doesn't connectionInfo already have host?

It doesn't https://github.com/gitpod-io/gitpod-vscode-desktop/blob/064833275fc9e65187a12a8a5cefb442ddc3befd/src/remote.ts#L10-L16